### PR TITLE
696 update individual profile view

### LINKED ISSUE #696: CHW/ANC - Update Individual  Profile view
1. No CarePlan to individual or Family 
Screen Shot 2021-11-01 at 12 43 36 PM

2. CarePlan referenced to individual or Family 

**On Course** 
Screen Shot 2021-11-01 at 12 43 36 PM

**Overdue**
Screen Shot 2021-11-01 at 1 02 29 PM

**See all Services**
Screen Shot 2021-11-10 at 9 23 34 AM

**View Past Encounters** 
Screen Shot 2021-11-10 at 9 21 19 AM

Screen Shot 2021-11-10 at 9 23 48 AM

**Acceptance Criteria** 
- [ ]  On create a Family or ANC member view a blank Profile page as viewed in mocks 
- [ ]  View location name and ID
- [ ]  Show head of the household tag if in Head profile  
- [ ] Create two sample patients on HAPI with referenced CarePlans one due and the other on-course 
- [ ] View patient profile for either Due or on Course 

**NB.**
1. All Upcoming services will be tracked as [Tasks](https://www.hl7.org/fhir/task.html#Task) references to a CarePlan. The task status will be **requested** 
2. All fulfilled [Tasks](https://www.hl7.org/fhir/task.html#Task) will be marked and referenced to an encounter. The task status will be **received** 
3. All [Encounters](https://www.hl7.org/fhir/encounter.html) will be referenced to a Patient or Head of Household 
4. A [CarePlan](https://www.hl7.org/fhir/careplan.html#CarePlan) will have a tag to whether it belongs to a patient or a household to be viewed in either the Family or Individual profile  

_You can create dummy resources to mock and reproduce the flow_

### PR DESCRIPTION
**Pull request by Rkareko**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #696 

**Type**
Choose one: Enhancement

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/anc/src/main/java/org/smartregister/fhircore/anc/data/patient/PatientRepository.kt [hunk 1]
```diff
@@ -303,13 +307,28 @@ class PatientRepository(
     )
   }
 
-  fun fetchUpcomingServiceItem(carePlan: List<CarePlan>): List<UpcomingServiceItem> {
+  suspend fun fetchUpcomingServiceItem(carePlan: List<CarePlan>): List<UpcomingServiceItem> {
     val listCarePlan = arrayListOf<UpcomingServiceItem>()
     val listCarePlanList = arrayListOf<CarePlan>()
     if (carePlan.isNotEmpty()) {
       listCarePlanList.addAll(carePlan.filter { it.due() })
       for (i in listCarePlanList.indices) {
```
Comment (ellykits) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/data/patient/PatientRepository.kt:315]: This can be converted to a `foreach` statement to make it more readable

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncDetailsFragment.kt [hunk 2]
```diff
@@ -155,20 +151,28 @@ class AncDetailsFragment : Fragment() {
...
 
   private fun populateImmunizationList(listCarePlan: List<CarePlanItem>) {
-    carePlanAdapter.submitList(listCarePlan)
+    val countOverdue = listCarePlan.filter { it.overdue }.size
+    val countDue = listCarePlan.filter { it.due }.size
+    if (countOverdue > 0) {
+      binding.txtViewCarePlan.text = "+ANC Visit $countOverdue Overdue"
+      binding.txtViewCarePlan.setTextColor(resources.getColor(R.color.status_red))
+    } else if (countDue > 0) {
+      binding.txtViewCarePlan.text = "+ANC Visit "
```
Comment (ellykits) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncDetailsFragment.kt:173]: Extract **+ANC Visit** to a string resource.

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncDetailsFragment.kt [hunk 3]
```diff
@@ -155,20 +151,28 @@ class AncDetailsFragment : Fragment() {
...
         binding.imageViewSeeAllArrow.visibility = View.GONE
-        binding.carePlanListView.visibility = View.GONE
+        binding.txtViewCarePlan.visibility = View.GONE
       }
       else -> {
         binding.txtViewNoCarePlan.visibility = View.GONE
         binding.txtViewCarePlanSeeAllHeading.visibility = View.VISIBLE
         binding.imageViewSeeAllArrow.visibility = View.VISIBLE
-        binding.carePlanListView.visibility = View.VISIBLE
+        binding.txtViewCarePlan.visibility = View.VISIBLE
```
Comment (ellykits) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncDetailsFragment.kt:160]: ENHANCEMENT (for readability): There are two extension methods for the `View` class that can be used to `show` or `hide` a view.

```
        binding.apply {
              txtViewNoCarePlan.hide()
              txtViewCarePlanSeeAllHeading.show()
              imageViewSeeAllArrow.show()
              txtViewCarePlan.show()
         }
````

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncDetailsFragment.kt [hunk 4]
```diff
@@ -155,20 +151,28 @@ class AncDetailsFragment : Fragment() {
         binding.txtViewNoCarePlan.visibility = View.VISIBLE
         binding.txtViewCarePlanSeeAllHeading.visibility = View.GONE
         binding.imageViewSeeAllArrow.visibility = View.GONE
-        binding.carePlanListView.visibility = View.GONE
+        binding.txtViewCarePlan.visibility = View.GONE
```
Comment (ellykits) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncDetailsFragment.kt:154]: You may perhaps consider using the View extension methods `show` or `hide` and use the `apply` scope function to enhance the code readability.

```kotlin
binding.apply {
        txtViewNoCarePlan.show()
        txtViewCarePlanSeeAllHeading.hide()
        imageViewSeeAllArrow.hide()
        txtViewCarePlan.hide()
}
```

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/PatientDetailsActivity.kt [hunk 5]
```diff
@@ -159,11 +159,16 @@ class PatientDetailsActivity : BaseMultiLanguageActivity() {
...
-      val patientId =
-        listOf(this.patientDetailsHead.demographics, this.patientDetails.patientIdentifier)
-          .joinToString(separator = " ID: ")
+      val patientAddress =
+        if (this.patientDetailsHead.demographics.isNotBlank()) this.patientDetailsHead.demographics
+        else this.patientDetails.demographics
+      val houseHoldHeadText =
+        if (this.patientDetails.isHouseHoldHead) getString(R.string.head_of_household) else ""
+      val patientIdText =
+        listOf(patientAddress, "ID: " + this.patientDetails.patientIdentifier, houseHoldHeadText)
```
Comment (ellykits) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/PatientDetailsActivity.kt:168]: You can refactor the string concatenation to use string interpolation.
`"ID: ${this.patientDetails.patientIdentifier}"`

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncDetailsFragment.kt (hunk 2 above)
Comment (ellykits) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncDetailsFragment.kt:173]: Extract **+ANC Visit** to a string resource.

**Review by ellykits — CHANGES_REQUESTED** [review]

**Review by ellykits — CHANGES_REQUESTED** [review]
Left a few comments and suggestions to improve code readability.

**Comment by Rkareko** [comment]
@ellykits PR feedback has been attended to

**Review by ellykits — APPROVED** [review]
